src/features/items.ts

```diff
@@ -1,175 +1,141 @@
-let itemsCounter = 0;
+import { reload } from "./loading";
+import { popSnack } from "./snack";
 
 export const LOAD_ITEMS = "LOAD_ITEMS";
 export type LoadItemsAction = {
   type: typeof LOAD_ITEMS;
+  items: ItemType[];
 };
-export const loadItems: () => LoadItemsAction = () => ({ type: LOAD_ITEMS });
+export const loadItems: (items: ItemType[]) => LoadItemsAction = (items) => ({
+  type: LOAD_ITEMS,
+  items,
+});
 
-export const DELETE_ITEM = "DELETE_ITEM";
-export type DeleteItemAction = {
-  type: typeof DELETE_ITEM;
-  listItem: ItemType;
+export const ITEM_DELETED = "ITEM_DELETED";
+export type ItemDeletedAction = {
+  type: typeof ITEM_DELETED;
+  item: ItemType;
 };
 
-export const deleteItem: (item: ItemType) => DeleteItemAction = (item) => {
+export const itemDeleted: (item: ItemType) => ItemDeletedAction = (item) => {
   return {
-    type: DELETE_ITEM,
-    listItem: item,
+    type: ITEM_DELETED,
+    item,
   };
 };
-export const ADD_NEW_ITEM = "ADD_NEW_ITEM";
-export type AddNewItemAction = {
-  type: typeof ADD_NEW_ITEM;
-  listItem: ItemType;
+
+export const ITEM_ADDED = "ITEM_ADDED";
+export type ItemAddedAction = {
+  type: typeof ITEM_ADDED;
+  item: ItemType;
 };
-export const addNewItem: (url: string) => AddNewItemAction = (url) => {
-  itemsCounter++;
+export const itemAdded: (item: ItemType) => ItemAddedAction = (item) => {
   return {
-    type: ADD_NEW_ITEM,
-    listItem: {
-      id: itemsCounter.toString(),
-      url,
-      date: new Date().getTime().toString(),
-      alreadyRead: false,
-      tags: [],
-    },
+    type: ITEM_ADDED,
+    item,
   };
 };
 
-export const ASSIGN_TAG_TO_ITEM = "ASSIGN_TAG_TO_ITEM";
-export type AssignTagToItemAction = {
-  type: typeof ASSIGN_TAG_TO_ITEM;
+export const TAG_ASSIGNED = "TAG_ASSIGNED";
+export type TagAssignedAction = {
+  type: typeof TAG_ASSIGNED;
   item: ItemType;
-  tag: TagType;
 };
-export const assignTagToItem: (
-  tag: TagType,
-  item: ItemType
-) => AssignTagToItemAction = (tag, item) => {
+export const tagAssigned: (item: ItemType) => TagAssignedAction = (item) => {
   return {
-    type: ASSIGN_TAG_TO_ITEM,
+    type: TAG_ASSIGNED,
     item,
-    tag,
   };
 };
 
-export const UNASSIGN_TAG_FROM_ITEM = "UNASSIGN_TAG_FROM_ITEM";
-export type UnassignTagFromItemAction = {
-  type: typeof UNASSIGN_TAG_FROM_ITEM;
-  tag: TagType;
+export const TAG_UNASSIGNED = "TAG_UNASSIGNED";
+export type TagUnassignedAction = {
+  type: typeof TAG_UNASSIGNED;
   item: ItemType;
 };
-export const unassignTagFromItem: (
-  tag: TagType,
-  item: ItemType
-) => UnassignTagFromItemAction = (tag, item) => {
+export const tagUnassigned: (item: ItemType) => TagUnassignedAction = (
+  item
+) => {
   return {
-    type: UNASSIGN_TAG_FROM_ITEM,
-    tag,
+    type: TAG_UNASSIGNED,
     item,
   };
 };
 
-export const TOGGLE_ITEM_STATUS = "TOGGLE_ITEM_STATUS";
-export type ToggleItemStatusAction = {
-  type: typeof TOGGLE_ITEM_STATUS;
+export const ITEM_TOGGLED = "ITEM_TOGGLED";
```

Comment:
Also I liked the name here before a bit better. `ITEM_TOGGLED` sounds quite generic and I think the `STATUS` part in that action description helped to understand what did toggle. Imagine you have a read/unread & visible/invisible based on some filter. I know we don't have that now but still I'd be happy to be more verbose here.